src/NuGet.Core/NuGet.Commands/NuGetLockFileBuilder.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Shared;
+
+namespace NuGet.Commands
+{
+    public class NuGetLockFileBuilder
+    {
+        private readonly int _lockFileVersion;
+        public NuGetLockFileBuilder(int lockFileVersion)
+        {
+            _lockFileVersion = lockFileVersion;
+        }
+
+        public NuGetLockFile CreateNuGetLockFile(LockFile assetsFile)
```

Comment:
can we call NuGetLockFile RepeatableBuildLockFile or something more descriptive? I do agree that it's incredibly confusing to see LockFile at someplace and NuGetLockFile in others.

Reply:
It'll be difficult to rename the current lock file classes, as they are used heavily in the build targets. 

Maybe in 5.0, but even then we'd likely have to take the initiative to fix the build targets ourselves.

Reply:
@nkolev92 so you are suggesting that we should just keep piling on the tech debt? Honestly, I can see how someone looking at this for the first time get really confused about the fact that there is a `LockFile` class that has nothing to do with the repeatable build lock file. I would prefer to fix that first before adding more debt that will later be more difficult to fix.

Reply:
I'm suggesting that we should have a naming concerns discussion and that this specific class LockFile is unlikely to change in 15.9 because the build targets are using it. 

I'd be up for cleaning up the APIs in 5.0/16.0

Reply:
I agree with @nkolev92 that we can discuss the naming concerns offline and come to a conclusion. 

But just to add my point of view here, we have some shitty naming conventions across our code base and I certainly wouldn't want to pile on top of that but we also have to realize that we're not in a re-factoring phase here and have the liberty to spend weeks and months on that. What we can do here is have less confusing names and better class summaries which clearly differentiates.

Now about this new lock file, then may I propose `PackagesLockFile` class name since the file name itself is `packages.lock.json` so it should be convenient to understand and have the clear summary to indicate that this class represents repeatable build `packages.lock.json` file.

Reply:
As discussed offline individually with most of you, I'll go ahead with `PackagesLockFile` with appropriate summary description and for all other subsequent classes. And also logged a tracking issue# https://github.com/NuGet/Home/issues/7207 to do some refactoring which we can discuss when we plan for dev16.